src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer_IdRangeAndCategoryValidation.cs

```diff
@@ -351,34 +354,39 @@ private static bool TryParseIdRangeEntry(string entry, out string prefix, out in
         {
             // Parse an entry for diagnostic ID.
             // We require diagnostic ID to have an alphabetical prefix followed by a numerical suffix.
-            prefix = string.Empty;
+            var prefixBuilder = new StringBuilder();
             suffix = -1;
-            string suffixStr = string.Empty;
+            var suffixStr = new StringBuilder();
             bool seenDigit = false;
             foreach (char ch in entry)
             {
                 bool isDigit = char.IsDigit(ch);
                 if (seenDigit && !isDigit)
                 {
+                    prefix = prefixBuilder.ToString();
                     return false;
                 }
 
                 if (isDigit)
                 {
-                    suffixStr += ch;
+                    suffixStr.Append(ch);
                     seenDigit = true;
                 }
                 else if (!char.IsLetter(ch))
                 {
+                    prefix = prefixBuilder.ToString();
                     return false;
                 }
                 else
                 {
-                    prefix += ch;
+                    prefixBuilder.Append(ch);
                 }
             }
 
-            return prefix.Length > 0 && suffixStr.Length > 0 && int.TryParse(suffixStr, out suffix);
+            prefix = prefixBuilder.ToString();
+            return prefix.Length > 0 &&
+                suffixStr.Length > 0 &&
```

Comment:
This check looks redundant given the next `TryParse` check?

Reply:
Yes, I will double check quickly the impl of `TryParse` just to make sure that's not some kind of optimization linked to the knowledge of how `TryParse` works.

Reply:
I had a look and I'd say both options are fine by me. The length check seems relevant given the implementation of `TryParse` but isn't functionally required.